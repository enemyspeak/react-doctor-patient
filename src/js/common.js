import React, { Component } from 'react';

function Avatar(props) {
	return (
		<img className="Avatar"
  			src={props.user.profile_image_url_https}
  			alt={props.user.screen_name}
		/>
	);
}

function UserInfo(props){
    return(
        <div className="tweet-user">
            <div className="user-avatar" onClick={() =>{if (props.onClick) props.onClick(props.user.screen_name)}}>
                <Avatar user={props.user} />
            </div>
            {props.user.verified && ( <div className="verified"></div> )}
            <span className="name" onClick={() =>{if (props.onClick) props.onClick(props.user.screen_name)}}>{props.user.name}</span>
            <span className="screenName" onClick={() =>{if (props.onClick) props.onClick(props.user.screen_name)}}>@{props.user.screen_name}</span>
        </div>
    );
}

function VideoMedia(props) {
	// TODO load video's bitrate responsive to size of viewport
	let index = 0;
	const variants = props.obj.video_info.variants;
	const bitrate_max = 832000;
	// const viewport = 300;
	const loop = (props.obj.type === "animated_gif" ? true : false );

	for (var i = variants.length - 1; i >= 0; i--) {
		if ((variants[i].content_type === "video/mp4") && (variants[i].bitrate < bitrate_max)) {
			index = i;
		}
	}
	return (
		<video controls loop={loop} poster={props.obj.media_url_https + ":small"} src={variants[index].url} type={variants[index].content_type} >
			{ /* {obj.video_info.variants.map(source => {
				return (
		    		<source src={source.url} type={source.content_type} />
		    	);
		    } */}
		</video>
	)
}

function Media(props) {
	// TODO load image size responsive to size of viewport
	// const viewport = 300;
	// const length = props.media.length;
	return (
		<div className="media-contain">
	        {props.media.map(obj => {
            	return (
            		<div className="media-box" key={obj.id_str}>
						{(obj.type === "video" || obj.type === "animated_gif") && (
							<VideoMedia obj={obj} />
						)}
						{(obj.type === "photo") && (
							<img src={obj.media_url_https + ":small"} alt={obj.display_url} />	
						)}
					</div>
				);
	        })}
		</div>
	);
}

class RelativeTime extends Component {
	relativeTime() {
		// console.log(this.props);
		let time = this.props.created_at;

	 	if (!time) return;

	    // let day,month,year;
	    let date = new Date(time),
	        diff = ( (( new Date().getTime() ) - date.getTime()) / 1000),
	        // day_diff = ( new Date(new Date().getFullYear(),new Date().getMonth(),new Date().getDate()).getTime() - new Date(date.getFullYear(),date.getMonth(),date.getDate()).getTime() ) / 1000 / 86400, //
	        day_diff = Math.floor(diff / 86400);
	    
	    if (isNaN(day_diff) || diff <= 0)
	        return (
	        	"now"
	            // year.toString()+'-'+
	            // ((month<10) ? '0'+month.toString() : month.toString())+'-'+
	            // ((day<10) ? '0'+day.toString() : day.toString())
	        );
	    
	    var r = (
	        diff > 0 &&
	        (
	        	// eslint-disable-next-line
	            day_diff === 0 &&
	            (
	                (
	                    (diff < 60 && Math.ceil(diff) + "s") ||
	                    (diff < 3600 && Math.ceil(diff / 60)  + "m") ||
	                    (diff < 7200 && "1h") ||
	                    (diff < 86400 && Math.floor(diff / 3600) + "h")
	                )
	            // eslint-disable-next-line
	            ) ||
	            (day_diff === 1 && "1d") ||
	            (Math.ceil(day_diff) + "d")
	        )
	    );
	    // console.log(r);
	    return r;
	}
   	render() {
   		// console.log(this);
		return (
			<span className="tweet-date">{this.relativeTime()}</span>
		) // 
   	}
}

class DropdownMenu extends Component {
	copyTextToClipboard(text) {
		var textArea = document.createElement("textarea");
		textArea.style.position = 'fixed';
		textArea.style.top = 0;
		textArea.style.left = 0;
		textArea.style.width = '2em';
		textArea.style.height = '2em';
		textArea.style.padding = 0;
		textArea.style.border = 'none';
		textArea.style.outline = 'none';
		textArea.style.boxShadow = 'none';
		textArea.style.background = 'transparent';
		textArea.value = text;

		document.body.appendChild(textArea);
		textArea.select();

		try {
			document.execCommand('copy');
			// var successful = document.execCommand('copy');
			// var msg = successful ? 'successful' : 'unsuccessful';
			// console.log('Copying text command was ' + msg);
		} catch (err) {
			console.log('Oops, unable to copy');
		}

		document.body.removeChild(textArea);

		this.props.hideMenu();
	}
	render() {
		if (this.props.tweet) {
			var text;
			if (this.props.tweet.extended_tweet) {
				text = this.props.tweet.extended_tweet.full_text;
			} else if (this.props.tweet.full_text) {
				text = this.props.tweet.full_text;
			} else {
				text = this.props.tweet.text;
			}
			return  (
				<div className={"tweet-menu " + (this.props.visible ? "visible" : "")}>
					<div className="list-item"><div className="list-icon fi-magnifying-glass"></div>Show Details</div>
					<div className="list-item" onClick={() => this.copyTextToClipboard("https://twitter.com/" + this.props.tweet.user.screen_name + "/status/" + this.props.tweet.id_str)}><div className="list-icon fi-quote"></div>Copy Link to Tweet</div>
					<div className="list-item" onClick={() => this.copyTextToClipboard(text)}><div className="list-icon fi-page-copy"></div>Copy Tweet Text</div>
					<a href={"https://twitter.com/" + this.props.tweet.user.screen_name + "/status/" + this.props.tweet.id_str} target="_blank" className="list-item"><div className="list-icon fi-link"></div>View on Twitter.com</a>
					{ this.props.mine && ( <div className="list-item"><div className="list-icon fi-delete"></div>Delete Tweet</div> )}
				</div>
			)
		}
		if (this.props.profile) {
			return  (
				<div className={"tweet-menu " + (this.props.visible ? "visible" : "")}>
					<div className="list-item" onClick={() => this.props.blockUser()}><div className="list-icon fi-prohibited"></div>Block User</div>
					<a href={"https://twitter.com/" + this.props.profile.screen_name } target="_blank" className="list-item"><div className="list-icon fi-link"></div>View on Twitter.com</a>
				</div>
			)
		}
	}
}

class StandardInput extends Component {
	constructor(props) {
    	super(props);
    	this.state = {
    		searchTerm: '',
    	};
    }
	handleChange(event) {
		this.setState({
			searchTerm: event.target.value,
		});
  	}
	render() {
		return (
			<div className="input-form-contain">
				<span className={"input-label " + (this.state.searchTerm ? "" : "full" )}>{this.props.inputName}</span>
				<input type={this.props.type || "text"} value={this.state.searchTerm} onChange={(event) => this.handleChange(event)} />
			</div>
		);
	}
}
export  { UserInfo,Avatar,Media,RelativeTime,DropdownMenu,StandardInput };