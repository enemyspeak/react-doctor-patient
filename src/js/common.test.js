import React from 'react';
import {Avatar,UserInfo,DropdownMenu,Media,VideoMedia} from './common';
import renderer from 'react-test-renderer';

const userData = {
	"id": 129175727,
	"id_str": "129175727",
	"name": "Gregorios Kythreotis",
	"screen_name": "ShedworksGreg",
	"location": "London, England",
	"description": "Cofounder + Design at Shedworks. Works with @ShedworksDan in a Shed. Architecture graduate.",
	"url": "https://t.co/ay4jDTPHV2",
	"entities": {
		"url": {
		  "urls": [
		    {
		      "url": "https://t.co/ay4jDTPHV2",
		      "expanded_url": "http://www.shed-works.co.uk",
		      "display_url": "shed-works.co.uk",
		      "indices": [
		        0,
		        23
		      ]
		    }
		  ]
		},
		"description": {
		  "urls": []
		}
	},
	"protected": false,
	"followers_count": 8623,
	"friends_count": 701,
	"listed_count": 190,
	"created_at": "Sat Apr 03 12:20:34 +0000 2010",
	"favourites_count": 18987,
	"utc_offset": 0,
	"time_zone": "London",
	"geo_enabled": true,
	"verified": false,
	"statuses_count": 8525,
	"lang": "en",
	"contributors_enabled": false,
	"is_translator": false,
	"is_translation_enabled": false,
	"profile_background_color": "C0DEED",
	"profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
	"profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
	"profile_background_tile": false,
	"profile_image_url": "http://pbs.twimg.com/profile_images/907574665543143425/8Ti_CjIB_normal.jpg",
	"profile_image_url_https": "https://pbs.twimg.com/profile_images/907574665543143425/8Ti_CjIB_normal.jpg",
	"profile_banner_url": "https://pbs.twimg.com/profile_banners/129175727/1505217608",
	"profile_link_color": "ABB8C2",
	"profile_sidebar_border_color": "C0DEED",
	"profile_sidebar_fill_color": "DDEEF6",
	"profile_text_color": "333333",
	"profile_use_background_image": true,
	"has_extended_profile": true,
	"default_profile": false,
	"default_profile_image": false,
	"following": false,
	"follow_request_sent": false,
	"notifications": false,
	"translator_type": "none"
};

test('Avatar render matches snapshot', () => {
	const component = renderer.create(
 		<Avatar user={userData} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('UserInfo render matches snapshot', () => {
	const component = renderer.create(
 		<UserInfo 
 			user={userData} 
 			onClick={(event) => console.log}  
 		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

const videomediadata = {
	// TODO
}

// test('Video Media render matches snapshot', () => {
// 	const component = renderer.create(
//  		<VideoMedia obj={videomediadata} />
// 	);
// 	let tree = component.toJSON();
// 	expect(tree).toMatchSnapshot();
// });

const mediadata = [
      {
        "id": 947842376537649200,
        "id_str": "947842376537649153",
        "indices": [
          64,
          87
        ],
        "media_url": "http://pbs.twimg.com/media/DSdpnpyWsAEcCBt.jpg",
        "media_url_https": "https://pbs.twimg.com/media/DSdpnpyWsAEcCBt.jpg",
        "url": "https://t.co/GyqXj0uTy2",
        "display_url": "pic.twitter.com/GyqXj0uTy2",
        "expanded_url": "https://twitter.com/ShedworksGreg/status/947842380098662402/photo/1",
        "type": "photo",
        "sizes": {
          "thumb": {
            "w": 150,
            "h": 150,
            "resize": "crop"
          },
          "large": {
            "w": 1353,
            "h": 2048,
            "resize": "fit"
          },
          "small": {
            "w": 449,
            "h": 680,
            "resize": "fit"
          },
          "medium": {
            "w": 793,
            "h": 1200,
            "resize": "fit"
          }
        },
        "source_status_id": 947842380098662400,
        "source_status_id_str": "947842380098662402",
        "source_user_id": 129175727,
        "source_user_id_str": "129175727"
      }
    ]

test('Media render matches snapshot', () => {
	const component = renderer.create(
 		<Media media={mediadata} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

const tweetData = 	{
	"id_str": "947972151574958081",
  	"full_text": "RT @ShedworksGreg: Spirit of the Mountain by Thomas Chamberlain https://t.co/GyqXj0uTy2",
  	user: {
	    "screen_name": "exoditedragon",
  	}
}

test('DropdownMenu tweet render matches snapshot', () => {
	const component = renderer.create(
	 	<DropdownMenu 
			visible={true}
			tweet={tweetData} 
			hideMenu={() => console.log}
		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('DropdownMenu tweet hidden render matches snapshot', () => {
	const component = renderer.create(
	 	<DropdownMenu 
			visible={false}
			tweet={tweetData} 
			hideMenu={() => console.log}
		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});


const profileData = {
	screen_name: "hannufluff"
}

test('DropdownMenu profile render matches snapshot', () => {
	const component = renderer.create(
	 	<DropdownMenu 
			visible={true}
			profile={profileData} 
			hideMenu={() => console.log}
		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('DropdownMenu profile hidden render matches snapshot', () => {
	const component = renderer.create(
	 	<DropdownMenu 
			visible={false}
			profile={profileData} 
			hideMenu={() => console.log}
		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});