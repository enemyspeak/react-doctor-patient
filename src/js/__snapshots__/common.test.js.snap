// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar render matches snapshot 1`] = `
<img
  alt="ShedworksGreg"
  className="Avatar"
  src="https://pbs.twimg.com/profile_images/907574665543143425/8Ti_CjIB_normal.jpg"
/>
`;

exports[`DropdownMenu profile hidden render matches snapshot 1`] = `
<div
  className="tweet-menu "
>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <div
      className="list-icon fi-prohibited"
    />
    Block User
  </div>
  <a
    className="list-item"
    href="https://twitter.com/hannufluff"
    target="_blank"
  >
    <div
      className="list-icon fi-link"
    />
    View on Twitter.com
  </a>
</div>
`;

exports[`DropdownMenu profile render matches snapshot 1`] = `
<div
  className="tweet-menu visible"
>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <div
      className="list-icon fi-prohibited"
    />
    Block User
  </div>
  <a
    className="list-item"
    href="https://twitter.com/hannufluff"
    target="_blank"
  >
    <div
      className="list-icon fi-link"
    />
    View on Twitter.com
  </a>
</div>
`;

exports[`DropdownMenu tweet hidden render matches snapshot 1`] = `
<div
  className="tweet-menu "
>
  <div
    className="list-item"
  >
    <div
      className="list-icon fi-magnifying-glass"
    />
    Show Details
  </div>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <div
      className="list-icon fi-quote"
    />
    Copy Link to Tweet
  </div>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <div
      className="list-icon fi-page-copy"
    />
    Copy Tweet Text
  </div>
  <a
    className="list-item"
    href="https://twitter.com/exoditedragon/status/947972151574958081"
    target="_blank"
  >
    <div
      className="list-icon fi-link"
    />
    View on Twitter.com
  </a>
</div>
`;

exports[`DropdownMenu tweet render matches snapshot 1`] = `
<div
  className="tweet-menu visible"
>
  <div
    className="list-item"
  >
    <div
      className="list-icon fi-magnifying-glass"
    />
    Show Details
  </div>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <div
      className="list-icon fi-quote"
    />
    Copy Link to Tweet
  </div>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <div
      className="list-icon fi-page-copy"
    />
    Copy Tweet Text
  </div>
  <a
    className="list-item"
    href="https://twitter.com/exoditedragon/status/947972151574958081"
    target="_blank"
  >
    <div
      className="list-icon fi-link"
    />
    View on Twitter.com
  </a>
</div>
`;

exports[`Media render matches snapshot 1`] = `
<div
  className="media-contain"
>
  <div
    className="media-box"
  >
    <img
      alt="pic.twitter.com/GyqXj0uTy2"
      src="https://pbs.twimg.com/media/DSdpnpyWsAEcCBt.jpg:small"
    />
  </div>
</div>
`;

exports[`UserInfo render matches snapshot 1`] = `
<div
  className="tweet-user"
>
  <div
    className="user-avatar"
    onClick={[Function]}
  >
    <img
      alt="ShedworksGreg"
      className="Avatar"
      src="https://pbs.twimg.com/profile_images/907574665543143425/8Ti_CjIB_normal.jpg"
    />
  </div>
  <span
    className="name"
    onClick={[Function]}
  >
    Gregorios Kythreotis
  </span>
  <span
    className="screenName"
    onClick={[Function]}
  >
    @
    ShedworksGreg
  </span>
</div>
`;
